kernel/src/main/java/org/kframework/backend/kore/ModuleToKORE.java

```diff
@@ -1451,7 +1451,7 @@ private boolean isGeneratedInKeysOp(Production prod) {
     }
 
     private Att addKoreAttributes(Production prod, SetMultimap<KLabel, Rule> functionRules, Set<Production> overloads) {
-        boolean isFunctional = !isFunction(prod);
+        boolean isFunctional = !isFunction(prod) || prod.att().contains(Att.TOTAL());
```

Comment:
Maybe this variable should be called `isTotalFunction` now.

Reply:
This is still an accurate name because that's what the attribubte is called at the kore level.